core/update.php

```diff
@@ -531,46 +622,109 @@ function update_check_requirements($skip_warnings = FALSE) {
 if (update_access_allowed()) {
 
   include_once BACKDROP_ROOT . '/core/includes/install.inc';
+  include_once BACKDROP_ROOT . '/core/includes/backup.inc';
   include_once BACKDROP_ROOT . '/core/includes/batch.inc';
   backdrop_load_updates();
 
   update_fix_compatibility();
 
+  $op = isset($_REQUEST['op']) ? $_REQUEST['op'] : '';
+  $valid_token = isset($_GET['token']) && backdrop_valid_token($_GET['token'], 'update');
+
   // Check the update requirements for all modules. If there are warnings, but
   // no errors, skip reporting them if the user has provided a URL parameter
   // acknowledging the warnings and indicating a desire to continue anyway. See
   // backdrop_requirements_url().
-  $skip_warnings = !empty($_GET['continue']);
-  update_check_requirements($skip_warnings);
+  if (!$op || $op == 'info') {
+    $skip_warnings = !empty($_GET['continue']);
+    update_check_requirements($skip_warnings);
+  }
 
-  $op = isset($_REQUEST['op']) ? $_REQUEST['op'] : '';
   switch ($op) {
     // update.php ops.
+    case 'info':
+      $output = update_info_page();
+      break;
+
+    case 'check_updates':
+      $update_count = update_get_update_count();
+      if ($update_count === 0) {
+        backdrop_set_message(t('No pending updates.') . ' ' . t('All caches cleared.'));
+
+        // No updates to run, so caches won't get flushed later.  Clear them now.
+        backdrop_flush_all_caches();
+        install_goto('core/update.php?op=results');
+      }
+      else {
+        // Skip the backup and go to update selection if upgrading from Drupal 7.
+        $op = update_backup_enabled() ? 'backup' : 'selection';
+        $token = backdrop_get_token('update');
+        install_goto('core/update.php?op=' . $op . '&token=' . $token);
+      }
+      break;
+
+    case 'backup':
+      if ($valid_token) {
+        $output = update_backup_page();
+      }
+      else {
+        install_goto('core/update.php');
+      }
+      break;
+
+    case t('Create backup'):
+      if ($valid_token) {
+        // Generate absolute URLs for the batch processing (using $base_root),
+        // since the batch API will pass them to url() which does not handle
+        // update.php correctly by default. Note the "action" query parameter
+        // here distinguishes between the two batch operations, which can be
+        // either "batch" or "update".
+        $batch_url = $base_root . backdrop_current_script_url(array('action' => 'backup'));
+        $batch_redirect_url = $base_root . backdrop_current_script_url(array('op' => 'selection'));
```

Comment:
Where does this `$base_root` variable come from? Can't seem to find it.

Reply:
It is a global defined in `backdrop_settings_initialize()`. I'm not much a fan of this URL creation but it's been in place for a long time.

Reply:
Oh, got it, we're not inside a function and this is one of the globals... Thank's for explaining. :+1: